jupyterhub/app.py

```diff
@@ -1410,7 +1410,13 @@ def init_internal_ssl(self):
     def init_db(self):
         """Create the database connection"""
 
-        self.log.debug("Connecting to db: %s", self.db_url)
+        #Do not log the entire DB connection string, strip the password
+        try:
+            db_info = ':'.join(self.db_url.split(':')[0:2]) + '@' + self.db_url.split('@')[1]
```

Comment:
Can you try using urlparse to make it more clear that we are redacting the password?

Something like:

```python
urlinfo = urlparse(self.db_url)
if urlinfo.password:
    # avoid logging the database password
    urlinfo = urlinfo._replace(netloc='{}:[redacted]@{}:{}'.format(urlinfo.username, urlinfo.hostname, urlinfo.port))
    db_log_url = urlinfo.geturl()
else:
    db_log_url = self.db_url
```